platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/manage/schema/extract/ScalarTypes.java

```diff
@@ -37,7 +38,8 @@ public abstract class ScalarTypes {
         ModelType.of(Double.class),
         ModelType.of(BigInteger.class),
         ModelType.of(BigDecimal.class),
-        ModelType.of(File.class)
+        ModelType.of(File.class),
+        ModelType.of(URI.class)
```

Comment:
💭 These types seem kind of not like the others.  What does `Scalar` even mean here?